pages/_app.js

```diff
@@ -32,27 +54,30 @@ export default function App({ Component, pageProps }) {
     );
   }
 
-  const shoppingItemsWithCategoryColor = shoppingItems.map((shoppingItem) => {
-    const category = categories.find(
-      (category) => category.name === shoppingItem.category
+  function handleToggleIsPurchased(id) {
+    setShoppingItems((prevItems) =>
+      prevItems.map((shoppingItem) =>
+        shoppingItem.id === id
+          ? { ...shoppingItem, isPurchased: !shoppingItem.isPurchased }
+          : shoppingItem
+      )
     );
-    const backgroundColor = category ? category.color : "white";
-    return {
-      ...shoppingItem,
-      backgroundColor,
-    };
-  });
+  }
 
   return (
     <>
       <GlobalStyle />
       <Component
         {...pageProps}
+        shoppingItems={shoppingItems}
+        categories={categories}
```

Comment:
Okay. 
--- Aber eine Frage bezüglich der Dateiendung: Sollten die Daten das Dateiformat .json haben? 
Bei mir sind es aktuell noch .js-Dateien.

Reply:
Thanks. Done.
I've got one question though: is it really preferable to import the categories in each file rather than to import them once and then pass them down?

Reply:
The files can be .json oder .js. You can leave them as they are.

Regarding your second question: I would say it always depends on the app. In your case, you just need the categories to display them and are not doing anything else with them (meaning, they always stay the same). In a case like that, you can just import them and do not need to pass them as props through multiple components.

Reply:
That makes sense. :)